src/numeric/decision_rules.jl

```diff
@@ -17,190 +51,169 @@ function filter_mcoeffs(a::Array{Float64,1},b::Array{Float64,1},n::Array{Int64,1
     return coeffs
 end
 
-# Constant Decision Rule
-type ConstantDecisionRule <: AbstractDecisionRule
-    values::Array{Float64,1}
-end
-(dr::ConstantDecisionRule)(x::Array{Float64,1}) = dr.values
-(dr::ConstantDecisionRule)(x::Array{Float64,2}) = repmat(dr.values', size(x, 1), 1)
-(dr::ConstantDecisionRule)(i::Int, x::Union{Vector,Matrix}) =  dr(x)
-(dr::ConstantDecisionRule)(i::Int, j::Int, x::Union{Vector, Matrix}) = dr(x)
 
 
-## Cubic spline decision rules
+#####
+##### 1-argument decision rule
+#####
 
-function set_values!(dr::AbtsractCubicSplineDR, values::Array{Array{Float64,2},1})
-    grid = dr.grid
-    a = grid.min
-    b = grid.max
-    orders = grid.n
-    dr.coefficients = [filter_mcoeffs(a, b, orders, v) for v in values]
-end
+using splines
 
-function set_values!(dr::AbtsractCubicSplineDR, values::Array{Float64,2})
-    set_values!(dr, [values])
+function DecisionRule(grid_exo::EmptyGrid, grid_endo::CartesianGrid, n_x::Int)
+    orders = grid_endo.n
+    coeffs = [zeros(n_x, (orders+2)...)]
+    return DecisionRule{EmptyGrid, CartesianGrid}(grid_exo, grid_endo, n_x, coeffs)
```

Comment:
So removing `{EmptyGrid, CartesianGrid}` doesn't worrk so I'm going to keep it until I understand what's going on...